### ISSUE TITLE
DataLad requires git configuration

### ISSUE THREAD

**Issue by djarecka** [datalad/datalad#2272]
#### What is the problem?
I'm creating a docker image with a dataset inside (using `pip` to install `datalad[full]`. It was working last week and  today I'm getting an error "[ERROR] You must configure git first (set both user.name and user.email) before using DataLad. [gitrepo.py:check_git_configured:349] (RuntimeError) "
I was wondering if this is an intended change.

#### What steps will reproduce the problem?

This is my [Dockerfile](https://github.com/miykael/nipype_tutorial/blob/master/Dockerfile) and here is [failing CircleCi](https://circleci.com/gh/miykael/nipype_tutorial/11)

This is [Circle CI report from last week](https://circleci.com/gh/djarecka/nipype_tutorial/47?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

**Comment by yarikoptic** [comment]
yes, this "Feature" came up with fresh 0.9.2 release from two days ago.  Some git operations really dislike having no git user.name/email configured. So now instead of waiting for things to break/work inconsistently we ask to configure git first

**Comment by yarikoptic** [comment]
I think we have rushed it a bit... man git-commit-tree describes:
```
       While parent object ids are provided on the command line, author and
       committer information is taken from the following environment
       variables, if set:

           GIT_AUTHOR_NAME
           GIT_AUTHOR_EMAIL
           GIT_AUTHOR_DATE
           GIT_COMMITTER_NAME
           GIT_COMMITTER_EMAIL
           GIT_COMMITTER_DATE

       (nb "<", ">" and "\n"s are stripped)

       In case (some of) these environment variables are not set, the
       information is taken from the configuration items user.name and
       user.email, or, if not present, the environment variable EMAIL, or, if
       that is not set, system user name and the hostname used for outgoing
       mail (taken from /etc/mailname and falling back to the fully qualified
       hostname when that file does not exist).
```
so us checking for config settings only is a bit "too crude".
I think, if config vars aren't defined do a quick check by issuing "git init; git commit" in a temp dir and detect complaints (I do not think it would be practical to track).  And then if we detect a problem we should either
- retain current behavior and demand configuration which still will not be friendly to "consumer" use cases as @djarecka where authorship on commits annex etc does isn't really important
- issue a warning, and proceed while defining a fake name/email variables (e.g. "DataLad User" for name and "datalad-user@example.com" for email) from aforementioned to be used during calls to git.  This way
@datalad/developers  what do you think?

**Comment by joeyh** [comment]
Note that git annex init detects when git commit needs hand-holding,
and sets user.name and user.email to the username. So datalad
might be able to avoid also needing to deal with this most annoying git
misfeature if it doesn't git commit before git annex init.

(I've had to work around this behavior of git so many times, I'd love
save anyone needing to work around it once more..)

-- 
see shy jo

**Comment by yarikoptic** [comment]
hm, what do you mean by "sets user.name and user.email"?  I see no such behavior (note thought that git manages to find my name from somewhere, not yet sure where from)
```shell
$> mv ~/.gitconfig{,.aside}

$> rm -rf /tmp/testrepo; export FULLNAME= NAME= EMAIL=; mkdir /tmp/testrepo; cd /tmp/testrepo; git init; git annex init ; git show git-annex; echo 123>123; git add 123; git commit -m msg
Initialized empty Git repository in /tmp/testrepo/.git/
init  ok
(recording state in git...)
commit 54e29af2d93f95ec351fb113320eaf99594bbd56 (git-annex)
Author: Yaroslav Halchenko <yoh@hopa.kiewit.dartmouth.edu>
Date:   Tue Mar 6 15:44:55 2018 -0500

    update

diff --git a/uuid.log b/uuid.log
new file mode 100644
index 0000000..0fc14bc
--- /dev/null
+++ b/uuid.log
@@ -0,0 +1 @@
+4bbba2fe-d086-4df3-ae5e-d7adca616878 yoh@hopa:/tmp/testrepo timestamp=1520369095.72274621s
[master (root-commit) 2d470ca] msg
 Committer: Yaroslav Halchenko <yoh@hopa.kiewit.dartmouth.edu>
Your name and email address were configured automatically based
on your username and hostname. Please check that they are accurate.
You can suppress this message by setting them explicitly:

    git config --global user.name "Your Name"
    git config --global user.email you@example.com

After doing this, you may fix the identity used for this commit with:

    git commit --amend --reset-author

 1 file changed, 1 insertion(+)
 create mode 100644 123
```

**Comment by joeyh** [comment]
It's done in Annex/Init.hs by ensureCommit if the git-annex branch
creation commit fails.

-- 
see shy jo

**Comment by yarikoptic** [comment]
addressed by #2299 -- should allow (and just warn) an unconfigured git -- things should still work in the basic set of usecases
